src/cli/serve/deprecated_config.js

```diff
@@ -0,0 +1,16 @@
+import { forOwn, has, noop } from 'lodash';
+
+// deprecated settings are still allowed, but will be removed at a later time. They
+// are checked for after the config object is prepared and known, so legacySettings
+// will have already been transformed.
+export const deprecatedSettings = new Map([
+  [['server', 'xsrf', 'token'], 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.']
```

Comment:
Well, I'm using `Map` because `_.has()` starts by doing a `hasOwnProperty()` check, and therefore [this test fails](https://github.com/elastic/kibana/pull/6825/files#diff-eaa820c11235b6f5b7dca2a846511229R27). It seems really weird to me that the function would work for direct properties, expanded properties, but not partially expanded properties, and so I used `Map`.

If you think that it's work it to prevent the use of `Map()` and have this weird behavior I can live with it, but it's not really a cost to me. `Map()` is no longer foreign to me.